indigo/analysis/refs/provisions.py

```diff
@@ -441,13 +506,46 @@ def candidates():
         for el in candidates():
             # look for the first marker tag (usually <ref>) or <term>
             if el.tag == self.marker_tag and el.get('href'):
-                return self.resolve_target(node, el.get('href'))
+                return self.resolve_target_uri(node, el.get('href'))
             elif self.ns and el.tag == "{%s}%s" % (self.ns, "term") and el.get('refersTo'):
-                return self.resolve_target(node, el.get('refersTo'))
+                return self.resolve_target_uri(node, el.get('refersTo'))
 
         return None, None
 
-    def resolve_target(self, node: Element, frbr_uri: str) -> Tuple[Union[str, None], Union[Element, None]]:
+    def get_target_from_text(self, match: re.Match, target: Union[str, None]) -> (Union[str, None], Union[Element, None]):
+        """Determine and resolve a remote target and return the target_frbr_uri and the appropriate root XML element of
+        the document to use to resolve the references. References to the local document are not supported because
+        the local document is assumed to be plain text.
+        """
+        if not target or target == "this":
```

Comment:
do these – 'this', 'thereof', 'of' – mean text ref extraction only works on English documents?

Reply:
No, they're magic constants that the grammar returns to indicate particular targets types. The grammar's alternative languages can use whatever they need to detect the various conditions, and then return these magic constants.

Reply:
oh amazing! I think I was thrown by them being strings, but that makes sense and is very cool!